solver/src/solver/oneinch_solver.rs

```diff
@@ -113,10 +113,11 @@ impl OneInchSolver {
         tracing::debug!("querying 1Inch swap api with {:?}", query);
         let swap = self.client.get_swap(query).await?;
 
-        ensure!(
-            swap.to_token_amount >= order.buy_amount,
-            "order limit price not respected",
-        );
+        if !swap.to_token_amount >= order.buy_amount {
```

Comment:
This is a pretty bad bug - it's not the same `!(swap.to_token_amount >= order.buy_amount)`